flink-core/src/main/java/org/apache/flink/api/common/functions/util/AbstractRuntimeUDFContext.java

```diff
@@ -97,10 +97,16 @@ public int getNumberOfParallelSubtasks() {
 	}
 
 	@Override
+	public int getMaxNumberOfParallelSubtasks() {
+		// Number of KeyGroups is same with max-parallelism currently.
+		return taskInfo.getNumberOfKeyGroups();
```

Comment:
what does this return for the batch API, which doesn't have key-groups? (ideally it should ```getNumberOFParallelSubtasks()```.